### PR TITLE
Update typos config

### PR DESCRIPTION
**Pull request by riemass**
This updates the extend list and solves issues that where detected with typos version 1.20.0.

### CODE REVIEW

File: .typos-config.toml [hunk 1]
```diff
@@ -3,4 +3,7 @@ extend-exclude = ["*.pem", "libcaf_net/test/pem.cpp", "Jenkinsfile"]
 
 [default.extend-words]
 caf = "caf"
-CAF = "CAF"
+aout = "aout"
+ws = "ws"
+yout = "yout"
```
Comment (Neverlord) [inline .typos-config.toml:8]: Typo? I couldn't find a match for `yout` (only for `layout`).

Comment (riemass) [reply .typos-config.toml:8]: That's for the template parameter Yout, e.g..

```
error: `Yout` should be `Your`
  --> ./libcaf_core/caf/interface_mismatch.hpp:77:26
   |
77 |         type_list<Zs..., Yout(Yin...)>> {};
   |                          ^^^^
```

Comment (Neverlord) [reply .typos-config.toml]: Ah, I didn't know it's matching case-insensitive.

**Comment by riemass** [comment]
I've capitalized `Yout` and sorted the `extend-words`. Squashed because it's a small PR anyway, there's no need for two commits.

**Pull request merged**